examples/advanced/Tutorial3/reconstruction/FairTestDetectorRecoTask.cxx

```diff
@@ -97,7 +97,9 @@ void FairTestDetectorRecoTask::Exec(Option_t* /*opt*/)
         TVector3 dpos(1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12));
 
         FairTestDetectorHit* hit = new ((*fHitArray)[ipnt]) FairTestDetectorHit(-1, -1, pos, dpos);
-        hit->AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), FairRootManager::Instance()->GetBranchId("FairTestDetectorDigi"), ipnt));
+        if ( FairRootManager::Instance() != NULL ) {
```

Comment:
`FairRootManager` has actually a strange implementation of the singleton pattern. Usually, `Instance()` returns a reference and then you don't need this check.

Btw, what is the meaning of your code, if you skip the `hit->AddLink` line? Is it really optional?